src/init.cpp

```diff
@@ -1363,9 +1363,16 @@ bool AppInitMain(InitInterfaces& interfaces)
     fDiscover = gArgs.GetBoolArg("-discover", true);
     fRelayTxes = !gArgs.GetBoolArg("-blocksonly", DEFAULT_BLOCKSONLY);
 
+    int listen_port = chainparams.GetDefaultPort();
+    assert(listen_port <= 0xffff);
+    listen_port = gArgs.GetArg("-port", listen_port);
+    if (listen_port > 0xffff) {
```

Comment:
nit: how about check against <1 values here? `src/qt/optionsdialog.cpp` does so:
https://github.com/bitcoin/bitcoin/blob/42653570373dad1fb431de2889764297d59ebcc0/src/qt/optionsdialog.cpp#L59

Reply:
@Empact Any reason why port 0 isn't valid?

Reply:
Not that I know of, for `proxyPort` it was disallowed in #1571.